apps/designer/app/routes/designer/$id.tsx

```diff
@@ -1,30 +1,69 @@
 import { useLoaderData } from "@remix-run/react";
-import { LoaderFunction } from "@remix-run/node";
+import {
+  ActionFunction,
+  LoaderFunction,
+  unstable_createFileUploadHandler,
+  unstable_parseMultipartFormData,
+} from "@remix-run/node";
 import type { Project } from "@webstudio-is/react-sdk";
 import { Designer, links } from "~/designer";
 import * as db from "~/shared/db";
 import config from "~/config";
 import env from "~/env.server";
+import { type Asset } from "@webstudio-is/react-sdk";
 export { links };
 
 export const loader: LoaderFunction = async ({ params }) => {
   if (params.id === undefined) throw new Error("Project id undefined");
   const project = await db.project.loadById(params.id);
+  const assets = await db.assets.loadByProject(params.id);
   if (project === null) {
     return { errors: `Project "${params.id}" not found` };
   }
-  return { config, project, env };
+  return { config, assets, project, env };
 };
 
+const directory = "./public/uploads";
+
 type Data = {
   config: typeof config;
   project: Project;
+  assets: Asset[];
 };
 
 type Error = {
   errors: "string";
 };
 
+type ImageUpload = {
+  type: string;
+  name: string;
+};
+
+export const action: ActionFunction = async ({ request, params }) => {
+  if (params.id === undefined) throw new Error("Project id undefined");
+  const formData = await unstable_parseMultipartFormData(
+    request,
+    unstable_createFileUploadHandler({
+      maxPartSize: 10_000_000,
+      directory,
+      file: ({ filename }) => filename,
+    })
+  );
+  const imageInfo = formData.get("image") as ImageUpload;
+  if (imageInfo) {
+    const data = {
+      type: imageInfo.type,
+      name: imageInfo.name,
+      path: `/uploads/${imageInfo.name}`,
```

Comment:
I think we should have fileName that would be `name + uuid` to prevent having clashes even if all files of all users end up in a single s3 storage folder some day

Reply:
have you logged imageInfo, whats inside?

Reply:
oh, remix does that by default, we do not need to do it, if an image exists they append the timestamp

Reply:
You can see that here: https://remix.run/docs/en/v1/api/remix#unstable_createfileuploadhandler-node